### PR TITLE
remove catalogsource switch logic

### PR DESCRIPTION
**Pull request by YCShen1010**
Signed-off-by: YCShen1010 <yuchen.shen@mail.utoronto.ca>

### CODE REVIEW

File: controllers/common/util.go [hunk 1]
```diff
@@ -503,141 +503,35 @@ func GetApprovalModeinNs(r client.Reader, ns string) (approvalMode string, err e
...
-		} else if pm.Status.CatalogSource == "redhat-operators" ||
-			pm.Status.CatalogSource == "redhat-marketplace" ||
-			pm.Status.CatalogSource == "community-operators" ||
-			pm.Status.CatalogSource == "certified-operators" {
+		} else if sub.Spec.name == "ibm-namespace-scope-operator" ||
+			sub.Spec.name == "ibm-odlm" {
 			continue
 		}
-		packageManifestList = append(packageManifestList, pm)
+		subscriptionList = append(subscriptionList, sub)
```
Comment (Daniel-Fan) [inline controllers/common/util.go:520]: Could we check `sub.Spec.name == packageName` and if it is yes, append it into `subscriptionList

File: controllers/common/util.go [hunk 2]
```diff
@@ -503,141 +503,35 @@ func GetApprovalModeinNs(r client.Reader, ns string) (approvalMode string, err e
...
-		r:                   r,
-		PackageManifestList: packageManifestList,
-	})
-
-	var packageManifestStringList []string
 
-	for _, packageManifest := range packageManifestList {
-		packageManifestStringList = append(packageManifestStringList, packageManifest.Status.CatalogSourceNamespace+"/"+packageManifest.Status.CatalogSource)
+	if len(subscriptionList) == 0 {
+		return  "", ""
```
Comment (Daniel-Fan) [inline controllers/common/util.go:524]: Could we raise an error msg `klog.Errorf()` here about no subscription found for ibm-common-service-oeprator?

File: controllers/common/util.go [hunk 3]
```diff
@@ -503,141 +503,35 @@ func GetApprovalModeinNs(r client.Reader, ns string) (approvalMode string, err e
...
 	}
 
-	klog.Infof("Available catalogsource for operator %v: %v", packageName, strings.Join(packageManifestStringList, ","))
-
-	if len(packageManifestList) == 0 {
+	if len(subscriptionList) > 1 {
+		klog.Infof("found more than one ibm-common-service-operator subscription in namespace: %v ", ns)
 		return "", ""
```
Comment (Daniel-Fan) [inline controllers/common/util.go:529]: Could we update the msg type from info to error?

Could we replace the `ibm-operator-service-operator` string in msg by `packageName` variable?

**Review by Daniel-Fan — COMMENTED** [review]
Hi @YCShen1010 
Please check the comments

**Review by qpdpQ — APPROVED** [review]

**Comment by Daniel-Fan** [comment]
Hold the PR since there are some error fixes required in the code

File: controllers/bootstrap/init.go [hunk 4]
```diff
@@ -168,6 +168,7 @@ func NewBootstrap(mgr manager.Manager) (bs *Bootstrap, err error) {
 		CSData:               csData,
 	}
 
+
```
Comment (Daniel-Fan) [inline controllers/bootstrap/init.go:171]: There is an extra line. Could we try to remove it?

**Comment by Daniel-Fan** [comment]
Hi @YCShen1010 

There are some format issues.
```
controllers/common/util.go:528: File is not `gofmt`-ed with `-s` (gofmt)

common/Makefile.common.mk:48: recipe for target 'lint-go' failed
make: *** [lint-go] Error 123
{"component":"entrypoint","error":"wrapped process failed: exit status 2","file":"prow/entrypoint/run.go:79","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","time":"2022-09-05T19:27:52Z"}
```

Could you please run the `make code-dev` cmd and [it will automatically fix the format error](https://github.com/IBM/ibm-common-service-operator/blob/ff4397a50cc8acd550113b5df4b7ef0b561c0fe1/Makefile#L157).

**Review by Daniel-Fan — APPROVED** [review]
/lgtm

**Pull request merged**